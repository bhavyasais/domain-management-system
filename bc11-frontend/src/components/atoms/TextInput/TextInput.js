import React from 'react';
import { withStyles, makeStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import { PLACEHOLDER } from '../../../utils/Strings';
import clsx from 'clsx';
import PropTypes from 'prop-types';

const TextFieldBox = withStyles((theme) => ({
  root: {
    '& .MuiOutlinedInput-root': {
      '& fieldset': {
        borderColor: theme?.palette?.lightGrey?.main,
      },
      '&:hover fieldset': {
        borderColor: theme?.palette?.lightGrey?.main,
      },
      '&.Mui-focused fieldset': {
        borderColor: theme?.palette?.lightGrey?.main,
      },
    },
    width: '360px',
    height: '52px',
  },
}))(TextField);

const useStyles = makeStyles((theme) => ({
  margin: {
    margin: theme.spacing(1),
    backgroundColor: theme?.palette?.white?.main,
  },
  labelProps: {
    width: '360px',
    height: '16px',
    fontSize: '14px',
    opacity: 0.6,
  },
}));
const TextInput = ({ value, onChange, inputStyle }) => {
  const classes = useStyles();
  return (
    <TextFieldBox
      className={clsx(classes.margin, inputStyle)}
      placeholder={PLACEHOLDER}
      variant="outlined"
      id="custom-css-outlined-input"
      inputProps={{ className: clsx(classes.labelProps) }}
      defaultValue={value}
      onChange={onChange}
    />
  );
};
TextInput.propTypes = {
  value: PropTypes.string,
};
TextInput.defaultProps = {
  value: { PLACEHOLDER },
};

export default TextInput;
