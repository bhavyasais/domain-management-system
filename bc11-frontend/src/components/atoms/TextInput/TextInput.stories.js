import React, { useState } from 'react';
import TextInput from './TextInput';

export default {
  title: 'Atoms/TextInput',
};

export const textInput = () => {
  const [state, setState] = useState("");
  const handleOnChange = (event) => {
    setState(event.target.value);
  };
  return <TextInput onChange={handleOnChange} value={state} />;
};
