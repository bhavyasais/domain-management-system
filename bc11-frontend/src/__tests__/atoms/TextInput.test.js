import React from 'react';
import renderer from 'react-test-renderer';
import TextInput from '../../components/atoms/TextInput/TextInput';

describe('TextInput', () => {
  it('should render correctly', () => {
    let wrapper;
    wrapper = renderer.create(<TextInput />);
    expect(wrapper).toMatchSnapshot();
  });
});
