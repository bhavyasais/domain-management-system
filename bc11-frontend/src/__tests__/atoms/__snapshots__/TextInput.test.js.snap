// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render correctly 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-3 makeStyles-margin-1"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input makeStyles-labelProps-2"
      defaultValue={
        Object {
          "PLACEHOLDER": "Mail@gmail.com",
        }
      }
      disabled={false}
      id="custom-css-outlined-input"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Mail@gmail.com"
      required={false}
      type="text"
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-5"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;
